subsys/net/lib/sockets/sockets.c

```diff
@@ -502,6 +501,7 @@ int zsock_poll(struct zsock_pollfd *fds, int nfds, int timeout)
 	int i;
 	int ret = 0;
 	struct zsock_pollfd *pfd;
+	struct k_poll_event poll_events[CONFIG_NET_SOCKETS_POLL_MAX];
```

Comment:
How much is this increasing stack usage, do you have any numbers?

Reply:
Well, "care" about stack usage was the reason it was originally a static array. And regardless of the stack usage, this is the right change to make. But if you ask, sizeof(struct k_poll_event) = 20, and CONFIG_NET_SOCKETS_POLL_MAX is by default set to 3.